aiida/storage/psql_dos/orm/groups.py

```diff
@@ -192,7 +193,7 @@ def check_node(given_node):
             if not given_node.is_stored:
                 raise ValueError('At least one of the provided nodes is unstored, stopping...')
 
-        with utils.disable_expire_on_commit(self.backend.get_session()) as session:
+        with self.backend.transaction() as session, disable_expire_on_commit(session) as session:
```

Comment:
Also, it feels that there are no tests actually checking this `disable_expire_on_commit` behaviour.
I assume it actually desired (I didn't add it), and if so then I think there should be a test for it

Reply:
Yeah, you are right, this is incorrect. And indeed, I don't think there is a proper test for this. If memory serves correctly, this was added way back when as a performance improvement. For a daemon worker, which could build up a session with quite a large number of instances in the session, it would invalidate them if just one node would be added to a group, which is not at all necessary.

I added a test that checks that a node instance, after adding it to a group, is not expired, however, the test passes with and without the `disable_expire_on_commit`. I am not sure whether I am using the SQLA API incorrectly to check whether an instance is expired, or if the way we are using the transaction, the node would anyway not get expired. Could you give a quick look to the test and see if you see anything weird/wrong? Maybe we can just remove the `disable_expire_on_commit`?

And if we do need to keep it, do you know how to make it work? I cannot do
```python
with disable_expire_on_commit(self.backend.transaction()) as session
```
because then `disable_expire_on_commit` would receive a `GeneratorContext` instead of a `Session`. How do we call `disable_expire_on_commit` while still passing the `session` of the `transaction` call?